### ISSUE TITLE
`export default class` emits non-default export

### ISSUE THREAD

**Issue by Arnavion** [Microsoft/TypeScript#2537]
This compiles and the output is correct.

```typescript
class One {
    foo() {
        return this;
    }
}

export default One;
```

```javascript
var One = (function () {
    function One() {
    }
    One.prototype.foo = function () {
        return this;
    };
    return One;
})();
exports.default = One;
```

---

Combining the class declaration and the export compiles, but the output is incorrect:

```typescript
export default class One {
    foo() {
        return this;
    }
}
```

```javascript
var One = (function () {
    function One() {
    }
    One.prototype.foo = function () {
        return this;
    };
    return One;
})();
exports.One = One;
```

The `default` is ignored.

**Comment by jeremyalan** [comment]
This issue appears to be happening again in v1.7.5... it works fine in v1.7.3.

Should this be reported on a new issue?

**Comment by jeremyalan** [comment]
This appears to be broken again in 1.7... it appears to be working fine in 1.6.

Should this be reported as a new issue?

**Comment by vladima** [comment]
Can you please share a repro code since I cannot reproduce this issue locally?
This is what I got using the latest bits from master:
```ts
"use strict";
var One = (function () {
    function One() {
    }
    One.prototype.foo = function () {
        return this;
    };
    return One;
}());
Object.defineProperty(exports, "__esModule", { value: true });
exports.default = One;
```

Looks like an expected output to me.

**Comment by jeremyalan** [comment]
I will try to post one when I get back to my computer... In the meantime, make sure you set `--target es6 --module commonjs`

**Comment by Arnavion** [comment]
Yes, it's fine in master but 1.7.5 gives

```javascript
class One {
    foo() {
        return this;
    }
}
exports.One = One;
```

**Comment by Arnavion** [comment]
Perhaps inadvertently fixed by https://github.com/Microsoft/TypeScript/commit/83e61cf that changed emitClassLikeDeclarationForES6AndHigher to call emitExportMemberAssignment

**Comment by jeremyalan** [comment]
Got it, thanks for the info!
